src/Engine/Objects/Player.cpp

```diff
@@ -3072,6 +3083,16 @@ PLAYER_SKILL_LEVEL Player::GetActualSkillLevel(PLAYER_SKILL_TYPE uSkillType) {
     // Vanilla returned 0 for PLAYER_SKILL_MISC here, we return 1.
     PLAYER_SKILL_LEVEL skill_value = GetSkillLevel(uSkillType);
 
+    if (shouldSkillBeShared(uSkillType)) {
+        for (Player &player : pParty->pPlayers) {
+            if (&player != this && player.CanAct()) {
+                PLAYER_SKILL_LEVEL otherPlayerSkillValue =
+                    player.GetSkillLevel(uSkillType);
+                skill_value = std::max(skill_value, otherPlayerSkillValue);
+            }
+        }
+    }
```

Comment:
Discord - could there ever be an instance where the mastery and skill level from different players is used? IE player 1  has normal skill but level 10 player 2 has expert level 4 - could you get a expert level 10 out by combination?

Reply:
Yes there could be, fixing it soon.